app/uk/gov/hmrc/cataloguefrontend/search/IndexBuilder.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cataloguefrontend.search
+
+
+import uk.gov.hmrc.cataloguefrontend.connector.{RepoType, TeamsAndRepositoriesConnector}
+import uk.gov.hmrc.http.{HeaderCarrier, StringContextOps}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+import uk.gov.hmrc.cataloguefrontend.healthindicators.{routes => healthRoutes}
+import uk.gov.hmrc.cataloguefrontend.leakdetection.{routes => leakRoutes}
+import uk.gov.hmrc.cataloguefrontend.teams.{routes => teamRoutes}
+import uk.gov.hmrc.cataloguefrontend.whatsrunningwhere.{routes => wrwRoutes}
+import uk.gov.hmrc.cataloguefrontend.{routes => catalogueRoutes}
+
+import java.util.concurrent.atomic.AtomicReference
+
+case class SearchTerm(name: String,  link: String, linkType: String) {
+  lazy val term: String = IndexBuilder.normalizeTerm(name)
+}
+
+@Singleton
+class IndexBuilder @Inject()(teamsAndRepositoriesConnector: TeamsAndRepositoriesConnector)(implicit ec: ExecutionContext){
+
+  val cachedIndex = new AtomicReference[Seq[SearchTerm]]()
+
+  val hardcodedLinks = List(
+    new SearchTerm("dependency explorer",      catalogueRoutes.DependencyExplorerController.landing.url, "explorer") {
+      override lazy val term: String = "depex"
```

Comment:
this is potentially a list of search terms (space separated) which we just look for matches in? So should we call it `terms`?